cpp/autosar/test/rules/M5-14-1/test.cpp

```diff
@@ -23,4 +23,14 @@ void f3(bool b) {
 
   if (b || f2()) { // COMPLIANT, f2 has local side-effects
   }
+}
+
+int g1 = 0;
+int f4() { return g1++; }
+int f5() { return 1; }
+
+void f6() {
+  if (noexcept(f5()) &&noexcept(
```

Comment:
Can we cover the other cases as well to ensure we properly handle those?

Reply:
great suggestion, this let me know that actually using decval that way is also not possible! good catch!